apps/bfd-db-migrator/src/test/java/gov/cms/bfd/migrator/app/MigratorAppIT.java

```diff
@@ -0,0 +1,327 @@
+          hasExceptionLogLine,
+          "Did not have expected log line for migration exception; STDOUT:\n"
+              + appRunConsumer.getStdoutContents());
+    } catch (ConditionTimeoutException e) {
+      throw new RuntimeException(
+          "Migration application failed to start within timeout, STDOUT:\n"
+              + appRunConsumer.getStdoutContents(),
+          e);
+    }
+  }
+
+  /**
+   * Test when the migration app runs and there are two of the same file with the same version, the
+   * exit code is 2.
+   *
+   * @throws IOException if there is an issue starting the app
+   */
+  @Test
+  public void
+      testMigrationRunWhenSchemaFileDuplicateVersionExpectFailedExitCodeAndMigrationsNotRun()
+          throws IOException {
+    // Setup
+    ProcessBuilder appRunBuilder = createAppProcessBuilder(TEST_DIRECTORY.DUPLICATE_VERSION);
+    appRunBuilder.redirectErrorStream(true);
+    Process appProcess = appRunBuilder.start();
+
+    ProcessOutputConsumer appRunConsumer = new ProcessOutputConsumer(appProcess);
+    Thread appRunConsumerThread = new Thread(appRunConsumer);
+    appRunConsumerThread.start();
+
+    // Await start/finish of application
+    try {
+      Awaitility.await().atMost(Duration.ONE_MINUTE).until(() -> !appProcess.isAlive());
+
+      LOGGER.info(appRunConsumer.getStdoutContents());
+
+      // Verify results
+
+      // Ensure the script directory was found (which would cause a false positive failure)
+      assertFalse(
+          hasLogLine(appRunConsumer, "Skipping filesystem location"),
+          "Could not find path to test files");
+
+      assertEquals(2, appProcess.exitValue());
+
+      // Test flyway threw an exception by checking the log output
+      assertTrue(
+          hasLogLine(appRunConsumer, "FlywayException"),
+          "Did not find expected flyway exception STDOUT:\n" + appRunConsumer.getStdoutContents());
+      assertTrue(
+          hasLogLine(appRunConsumer, "Found more than one migration with version"),
+          "Did find duplicate version error STDOUT:\n" + appRunConsumer.getStdoutContents());
+    } catch (ConditionTimeoutException e) {
+      throw new RuntimeException(
+          "Migration application failed to start within timeout, STDOUT:\n"
+              + appRunConsumer.getStdoutContents(),
+          e);
+    }
+  }
+
+  /**
+   * Gets the number of migration scripts by checking the directory they are located in.
+   *
+   * @return the number migration scripts
+   */
+  public int getNumMigrationScripts() {
+    String migrationPath = "src/main/resources/db/migration";
+
+    long fileCount = 0;
+    try (Stream<Path> files = Files.list(Paths.get(migrationPath))) {
+      fileCount = files.count();
+    } catch (IOException io) {
+      fail("Could not find file path for migration scripts at: " + migrationPath);
+    }
+
+    assertNotEquals(
+        0, fileCount, "Found no migration scripts at the expected location: " + migrationPath);
```

Comment:
it is somewhat unexpected that a method called getXXX() is performing an assertion. At a minimum I would include this in the javadoc but it is more natural IMO for the caller to make the assertion especially since in theory it is perfectly fine to have a migration directory that is empty.

Reply:
I think I was originally thinking this would always be being used against the "real" script directory so something should always be there, but this could get reused so I'll just remove that; the caller can decide if 0 is acceptable.